packages/api/src/command/medical/patient/get-patient-facility-matches.ts

```diff
@@ -0,0 +1,105 @@
+import { Address } from "@metriport/core/domain/address";
+import { USState } from "@metriport/api-sdk";
+import { LinkDemographics } from "@metriport/core/domain/patient-demographics";
+import { CQLink } from "../../../external/carequality/cq-patient-data";
+import { CwLink } from "../../../external/commonwell/cw-patient-data";
+import { CQPatientDataModel } from "../../../external/carequality/models/cq-patient-data";
+import { CwPatientDataModel } from "../../../external/commonwell/models/cw-patient-data";
+import { CQDirectoryEntryModel } from "../../../external/carequality/models/cq-directory";
+import { orgDirectory } from "../../../external/commonwell/org-directory";
+import { patientResourceToNormalizedLinkDemographics as cqPatientResourceToNormalizedLinkDemographics } from "../../../external/carequality/patient-demographics";
+import { patientNetworkLinkToNormalizedLinkDemographics as cwPatientResourceToNormalizedLinkDemographics } from "../../../external/commonwell/patient-demographics";
+
+type PatientFacilityMatch = {
+  name: string;
+  oid: string;
+  address: Partial<Address>;
+  patient?: LinkDemographics;
+};
+
+export async function getPatientFacilityMatches({
+  patientId,
+}: {
+  patientId: string;
+}): Promise<PatientFacilityMatch[]> {
+  const [cqPatientData, cwPatientData] = await Promise.all([
+    CQPatientDataModel.findOne({ where: { id: patientId } }),
+    CwPatientDataModel.findOne({ where: { id: patientId } }),
+  ]);
+
+  const cqPatientDataLinks = cqPatientData?.data.links ?? [];
+  const cwPatientDataLinks = cwPatientData?.data.links ?? [];
+
+  const [cqPatientFacilityMatches, cwPatientFacilityMatches] = await Promise.all([
+    getCqFacilityMatches(cqPatientDataLinks),
+    getCwFacilityMatches(cwPatientDataLinks),
+  ]);
+
+  return [...cqPatientFacilityMatches, ...cwPatientFacilityMatches];
+}
+
+async function getCqFacilityMatches(cqLinks: CQLink[]): Promise<PatientFacilityMatch[]> {
+  const patientFacilityMatches: PatientFacilityMatch[] = [];
+
+  for (const cqLink of cqLinks) {
+    if (!cqLink.patientResource) {
+      continue;
+    }
+
+    const cqFacility = await CQDirectoryEntryModel.findOne({
+      where: { id: cqLink.oid },
+    });
+
+    if (!cqFacility) {
+      continue;
+    }
+
+    const patientMatchDemo = cqPatientResourceToNormalizedLinkDemographics(cqLink.patientResource);
+
+    patientFacilityMatches.push({
+      name: cqFacility.name ?? "",
+      oid: cqFacility.id ?? "",
```

Comment:
Why empty string as the fallback? If this was the Dashboard and we were to display it, I understand not wanting to render `undefined`. But everywhere else, I think we should use `undefined` to represent the lack of a value, not empty string.